internal/k8s/reconciler/route.go

```diff
@@ -205,6 +205,8 @@ func (r *K8sRoute) OnBindFailed(err error, gateway store.Gateway) {
 				status.ResolvedRefs.ConsulServiceNotFound = err
 			case service.K8sServiceResolutionErrorType:
 				status.ResolvedRefs.ServiceNotFound = err
+			case service.RefNotPermittedErrorType:
```

Comment:
Gah, I totally forgot about the need to do this here -- glad you figured it out. For some context for everyone's sake if you're wondering why the code is like this, basically we can't set the status of the route directly when it has something that's invalid.

"Why" you ask? Because according to the spec, routes only have statuses in relation to the parent refs that they specify (yes, this makes route statuses all the more confusing when something is invalid for a route globally, as is the case here).

The result? We memoize any resolution errors from our validation stage onto the struct that we store and then, once binding to a gateway inevitably fails, we dump the resolution errors onto the parentRef-based status.

Reply:
Another thing to note, the reason we can't just store statuses on all parent refs for the route prior to attempting the insertion is because during the reference resolution process we don't actually know which gateways that the route references are managed by our controller -- we're not allowed to touch statuses for gateways that aren't managed by us.